common/types/endpoint.go

```diff
@@ -343,3 +414,38 @@ func (e *Endpoint) InvalidatePolicy() {
 		e.Consumable.Iteration = 0
 	}
 }
+
+func (e *Endpoint) LogStatus(code StatusCode, msg string) {
```

Comment:
I'm fine with merging what you have for now and then address this in a follow-up commit.